File: mili-server/src/main/java/fun/bm/mili/config/modules/function/ReplayAPIConfig.java
package fun.bm.mili.config.modules.function;

import fun.bm.mili.rust.TomlConfigData;
import fun.bm.mili.utils.RandomProfilePool;
import me.earthme.luminol.config.IConfigModule;
import me.earthme.luminol.config.flags.ConfigClassInfo;
import me.earthme.luminol.config.flags.ConfigInfo;
import me.earthme.luminol.config.flags.HotReloadUnsupported;
import me.earthme.luminol.enums.EnumConfigCategory;
import org.jetbrains.annotations.Nullable;

import java.util.Set;

@ConfigClassInfo(category = EnumConfigCategory.FUNCTION, name = "replay-api")
public class ReplayAPIConfig implements IConfigModule {
    @HotReloadUnsupported
    @ConfigInfo(name = "enable-cache")
    public static boolean enableCache = true;

    @HotReloadUnsupported
    @ConfigInfo(name = "cache-photographer-time", comments = """
            缓存摄影师资料的时间（秒）""")
    public static int cachePhotographerTime = 3600;

    @HotReloadUnsupported
    @ConfigInfo(name = "cache-photographer-size", comments = """
            缓存摄影师资料的最大数量""")
    public static int cachePhotographerSize = 100;

    @Override
    public void onLoaded(TomlConfigData configInstance, @Nullable Set<Exception> e) {
        RandomProfilePool.init();
    }
}

File: mili-server/src/main/kotlin/fun/bm/mili/config/modules/function/ReplayAPIConfig.kt
package `fun`.bm.mili.config.modules.function
import com.electronwill.nightconfig.core.file.CommentedFileConfig
import `fun`.bm.mili.config.modules.ConfigModule
import me.earthme.luminol.config.flags.ConfigClassInfo
import me.earthme.luminol.config.flags.ConfigInfo
import me.earthme.luminol.enums.EnumConfigCategory

@ConfigClassInfo(category = EnumConfigCategory.FUNCTION, name = "replay-api")
class ReplayAPIConfig : ConfigModule {
    companion object {
        @ConfigInfo(name = "enabled", comments = "Enable Replay API support") @JvmField var enabled = false
        @ConfigInfo(name = "enable-cache", comments = "Enable photographer cache") @JvmField var enableCache = false
        @ConfigInfo(name = "cache-photographer-time", comments = "Cache photographer time") @JvmField var cachePhotographerTime = 300L
        @ConfigInfo(name = "cache-photographer-size", comments = "Cache photographer size") @JvmField var cachePhotographerSize = 100L
    }
    override fun onLoaded(c: CommentedFileConfig) {}
    override fun onUnloaded(c: CommentedFileConfig) {}
}